### PR TITLE
add trim() to addAddress

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

**Review by mwaliullah-dev — DISMISSED** [review]
It seems that adding trim breaks the app. I get internal server error, when I try to add an address. Here is the error:

lyway options: jdbc:postgresql://database-server:5432/database
app-1             | [uncaught application error]: TypeError - Cannot read properties of null (reading 'trim')
app-1             | 
app-1             | request: { url: "http://localhost:7777/", method: "POST", hasBody: true }   
app-1             | response: { status: 404, type: undefined, hasBody: false, writable: true } 

app-1             |     at addAddress (file:///app/routes/controllers/addressController.js:16:34

problem is at this line:   const name = params.get("name").trim();
Basically, because of trim. I can take a look at it later again. But please check your code, try to run it. 
But it looks like trim is being called on null value. Perhaps, you can add a null check before calling trim?

**Review by MarcMcBurney — APPROVED** [review]
Tested locally and works as expected

**Comment by ghost** [comment]
Thank you!

**Comment by ghost** [comment]
Hi!
   I still dont merge my branch, and I ask the pull request to Marc
McBurney (he dont have any problem), so I dont why you have problems with
my code. But the problem of your code is that you forget to add the
firstName and lastName to the controller. Probably you should modify the
service and the database too. When you remove the input name, the
controller cannot found it and return null (const name =
params.get("name");), when you add the inputs firstName and lastName,
you should modify the function const addAddress = async ({ request,
response }) ;
const firstName = params.get("firstName");
const lastName = params.get("lastName");
The add to modify the service too:
 await executeQuery(`INSERT INTO addresses (firstName,lastName, address)
And add the field in the table addresses.

Then when I try to merge my code on Monday I can modify my code to :
const firstName = params.get(" firstName ").trim(); //trim remove the white
spaces at the beggining and the end of the string
const lastName = params.get(" lastName ").trim();

If you do not complete your changes by Monday, I will merge and commit my
code, and you will have to adapt to my changes.
If you have any questions, ask Petteri.

Best regards,
      Hoakin

**Comment by mwaliullah-dev** [comment]
Hei Hoakin, 

I updated my codes and created a new PR.  I changed the addAddress function in addressControlle.js like this:

// Retrieving name and address from the form data.
  const lastName = params.get("lastName");
  const firstName = params.get("firstName");
  const streetAddress = params.get("streetAddress");
  const postCodeCity = params.get("postCodeAndCity");
  console.log(`Name: ${lastName} ${firstName}\nAddress: ${streetAddress}, ${postCodeCity}`);
  
  But I did not change addressService.js or the schemal. Because I combined the inputs to match the existing schema like this:
  
// Combine fields to match service requirements
  const name = `${firstName} ${lastName}`; // Full name
  const address = `${streetAddress}, ${postCodeCity}`; // Full address

I tested the code locally before committing, and it works as expected. Please review my code. 

Thank you so much. 

--Md Waliullah
